firebase/functions/src/payments.ts

```diff
@@ -0,0 +1,183 @@
+    }
+    await admin.firestore().collection("payments").doc(session.id).set({
+      sessionId: session.id,
+      createdAt: admin.firestore.FieldValue.serverTimestamp(),
+      amount,
+      email,
+      recurring: type === "recurring",
+    });
+
+    functions.logger.debug("payment initiate result", session);
+    return {
+      sessionId: session && session.id,
+    };
+  },
+  webhook: async (request: functions.Request, response: functions.Response) : Promise<any> => {
+    functions.logger.debug("payment webhook parameters", request.body);
+
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore
+    const payload = request.rawBody;
+    const sig = request.headers["stripe-signature"];
+
+    let event;
+
+    try {
+      event = stripe.webhooks.constructEvent(payload, sig, endpointSecret);
+    } catch (err) {
+      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+      // @ts-ignore
+      functions.logger.warn(`Stripe Webhook Error: ${err.message}`);
+      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+      // @ts-ignore
+      return response.status(400).send(`Webhook Error: ${err.message}`);
+    }
+
+    // Handle the checkout.session.completed event
+    if (event.type === "checkout.session.completed") {
+      const session = event.data.object;
+
+      if (session.mode === "setup") {
+        const priceId = session.metadata.price_id;
+
+        const setupIntentID = session.setup_intent;
+        const intent = await stripe.setupIntents.retrieve(setupIntentID);
+
+        const method = await stripe.paymentMethods.retrieve(intent.payment_method);
+        const billingDetails = method.billing_details;
+
+        const email = billingDetails.email;
+
+        /* Create customer */
+        const customer = await stripe.customers.create({
+          email,
+          name: billingDetails.name,
+          address: billingDetails.address,
+          payment_method: intent.payment_method,
+          invoice_settings: {
+            default_payment_method: intent.payment_method,
+          },
+        });
+
+        const customerId = customer.id;
+        await stripe.subscriptions.create({
+          customer: customerId,
+          items: [
+            {price: priceId},
+          ],
+        });
+        await admin.firestore().collection("payments").doc(session.id).update({
+          sessionId: session.id,
+          updatedAt: admin.firestore.FieldValue.serverTimestamp(),
+          completed: true,
+          email,
+          customerId,
+        });
+
+        // await sendThankYouEmail(email);
+
+        // const amount = {
+        //   "price_1IAgywKEl0DmQOIqsyiOEF88": "2,50",
+        //   "price_1IAgzIKEl0DmQOIqLGl1nBIA": "5,00",
+        //   "price_1IAgzbKEl0DmQOIqEuVJitsi": "10,00",
+        // }[priceId];
+        // await postSlackUpdateMember(amount);
```

Comment:
No commented out code please

Reply:
You can remove it, but I think it's nice to keep it there since it's all features you might want to add later.

Reply:
If there are future expansions to the donation feature you'd suggest, maybe create a ticket in github? More likely to be picked up then  a section of commented out code w/o context.

Reply:
I think those tickets would just get buried really, but commented code you will find as soon as you're looking how to extend some functionality or how something has been implemented. Commented code is just as useful as real comments. (But adding a comment to commented code on what it does would definitely make it a lot better)